subprojects/core/src/main/java/org/gradle/internal/service/scopes/VirtualFileSystemServices.java

```diff
@@ -190,36 +186,21 @@ WatchingAwareVirtualFileSystem createVirtualFileSystem(
                 updateFunctionDecorator,
                 DirectoryScanner.getDefaultExcludes()
             );
-            WatchingAwareVirtualFileSystem watchingAwareVirtualFileSystem = determineWatcherRegistryFactory(OperatingSystem.current())
-                .<WatchingAwareVirtualFileSystem>map(watcherRegistryFactory -> new WatchingVirtualFileSystem(
-                    watcherRegistryFactory,
-                    delegate,
-                    updateFunctionDecorator,
-                    watchFilter,
-                    sectionId -> documentationRegistry.getDocumentationFor("gradle_daemon", sectionId)
-                ))
-                .orElse(new NonWatchingVirtualFileSystem(delegate));
+            WatchingAwareVirtualFileSystem watchingAwareVirtualFileSystem = fileWatcherRegistryFactory instanceof UnavailableFileWatcherRegistryFactory
+                ? new NonWatchingVirtualFileSystem(delegate)
+                : new WatchingVirtualFileSystem(
+                fileWatcherRegistryFactory,
+                delegate,
+                updateFunctionDecorator,
+                watchFilter,
+                sectionId -> documentationRegistry.getDocumentationFor("gradle_daemon", sectionId)
+            );
             listenerManager.addListener((BuildAddedListener) buildState ->
                 watchingAwareVirtualFileSystem.buildRootDirectoryAdded(buildState.getBuildRootDir())
             );
             return watchingAwareVirtualFileSystem;
```

Comment:
The naming here looks fishy. We now sometimes create a VFS that is not really "watching aware" as it does not watch. And we add a listener that is not really necessary when you have a non-watching file system, too. I wonder if we could simplify this.